.github/workflows/packaging.yml

```diff
@@ -0,0 +1,38 @@
+# Uses PyPA's build tool to build a wheel and sdist.
+# See https://packaging.python.org/en/latest/guides/distributing-packages-using-setuptools/#packaging-your-project
+
+name: Build packages for distribution
+on:
+  workflow_call:
+
+
+jobs:
+  packaging:
+    runs-on: ubuntu-latest
+    name: Packaging
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Python
+        uses: actions/setup-python@v2
+
+      - name: Install tools
+        run: python -m pip install build twine
+
+      - name: Build wheel and sdist
+        run: python -m build
+
+      # This checks whether your distribution’s long description will render correctly on PyPI.
+      - name: Run twine check
+        run: python -m twine check dist/*
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: Wheel
+          path: dist/*.whl
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: Sdist
+          path: dist/*.tar.gz
```

Comment:
~~are we going to use this pipeline for Synapse eventually? Do we want this automatic upload for Synapse (it would change the current procedure).
I'm actually fairly happy to have it done automatically, but I guess it's something that could be checked with the team.~~

~~Depending on the answer, we might want this to be gated behind a condition / input flag.~~

~~Edit: this also applies to the wheel~~

Edit 2: Wait — is this uploading to GitHub or to PyPI? I now think this is GitHub, right? Installing twine beforehand confused me; if it's not just me being stupid, would it be OK to add a comment saying that this uploads to GitHub... :sweat_smile:

Reply:
This is just to github. Twine is there as a "check we could upload this wheel to PyPI if we wanted to".